src/pages/SignIn/components/UserSignIn.jsx

```diff
@@ -1,42 +1,79 @@
 import { signInWithEmailAndPassword } from 'firebase/auth';
-import React, { useRef } from 'react';
+import React, { useRef, useState } from 'react';
 import { auth } from '../../../utils/firebase';
 import { Input, Button } from '@nextui-org/react';
+import { EyeFilled, EyeInvisibleFilled } from '@ant-design/icons';
 
 const UserSignIn = () => {
   const email = useRef();
   const password = useRef();
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isInvalid, setIsInvalid] = useState(false);
+  const [isVisible, setIsVisible] = React.useState(false);
+
+  const toggleVisibility = () => setIsVisible(!isVisible);
+
   const signIn = (e) => {
     e.preventDefault();
     signInWithEmailAndPassword(auth, email.current.value, password.current.value)
       .then((userCredential) => {
         console.log(userCredential);
       })
       .catch((error) => {
-        console.log(error);
+        console.log(error.code);
+        const message = error.code.replace('-', ' ').replace('auth/', '');
+        console.log('message', message);
+        setErrorMessage(message);
+        setIsInvalid(true);
       });
   };
 
+  const handleInputClick = () => {
+    setIsInvalid(false);
+  };
+
   return (
     <div className="sign-in-container">
       <form id="myForm" onSubmit={signIn}>
         <div className="flex w-[400px] mx-auto flex-col flex-wrap mb-6 md:mb-0 gap-4">
           <Input
+            label="Email"
             size="md"
-            type="email"
             variant="bordered"
-            label="Email"
             isRequired={true}
             ref={email}
+            isInvalid={isInvalid}
+            color={isInvalid ? 'danger' : 'default'}
+            onClick={handleInputClick}
           />
           <Input
+            label="Password"
             size="md"
-            type="password"
             variant="bordered"
-            label="Password"
             isRequired={true}
             ref={password}
+            isInvalid={isInvalid}
+            color={isInvalid ? 'danger' : 'default'}
+            onClick={handleInputClick}
+            endContent={
+              <Button
+                className="focus:outline-none bg-transparent pt-2"
+                type="button"
+                size="sm"
+                disableRipple="true"
+                onClick={toggleVisibility}>
+                {isVisible ? (
+                  <EyeInvisibleFilled className="text-2xl text-default-400 pointer-events-none" />
+                ) : (
+                  <EyeFilled className="text-2xl text-default-400 pointer-events-none" />
+                )}
+              </Button>
+            }
+            type={isVisible ? 'text' : 'password'}
           />
+          <p className="text-left text-[#f31260]">
+            {errorMessage === 'invalid credential' && <p>Please enter valid credentials.</p>}
```

Comment:
why don't you just print errorMessage ? you already replaced error.code

Reply:
what about removing this and add `errorMessage={errorMessage}` in password Input?